internal/controllers/rbac.go

```diff
@@ -261,12 +268,41 @@ func (r *Reconciler) createOrUpdateRoleBinding(ctx context.Context, binding *rba
 		return nil
 	})
 	if err != nil {
+		if err == errRoleRefModified {
+			return r.recreateRoleBinding(ctx, bindingCopy, owner, subjects, roleRef)
+		}
 		return err
 	}
 	r.Log.Info(fmt.Sprintf("Role Binding %s", op), "name", binding.Name, "namespace", binding.Namespace)
 	return nil
 }
 
+var errRoleRefModified error = errors.New("role binding roleRef has been modified")
+
+func getRoleRef(binding metav1.Object, oldRef *rbacv1.RoleRef, newRef *rbacv1.RoleRef) (*rbacv1.RoleRef, error) {
+	// The RoleRef field is immutable. In order to update this field, we need to
+	// delete and re-create the role binding.
+	// See: https://kubernetes.io/docs/reference/access-authn-authz/rbac/#clusterrolebinding-example
+	creationTimestamp := binding.GetCreationTimestamp()
+	if creationTimestamp.IsZero() {
+		return newRef, nil
+	} else if !cmp.Equal(oldRef, newRef) {
+		// Return error so deployment can be recreated
```

Comment:
// Return error so ~deployment~ `rolebinding/clusterrolebinding` can be recreated?

Reply:
Fixed. Thanks for catching that